pkg/cmd/root.go

```diff
@@ -143,8 +144,18 @@ func ensureConfigDefaults() {
 	}
 
 	if needsWrite {
-		fmt.Println("updating configfile to include defaults")
+		spinner, err := ui.NewSpinner()
+		cobra.CheckErr(err)
+		err = spinner.Start()
+		cobra.CheckErr(err)
+		spinner.Message("Updating configfile to include defaults")
```

Comment:
This seems really verbose.
How about Rename spinner to subCommand? You could then also be able to deal with timeouts.
```golang
err = subcommand.Run(func (){
    return viper.WriteConfig()
}, subcommand.Opts{Start:"start message", End:"end message", WithSpinner: true, Timeout: 5*time.Second})
cobra.CheckErr(err)
```